### PR TITLE
Per-field validate function

### PR DESCRIPTION
**Pull request by estambakio-sc**
#44

### CODE REVIEW

File: src/data-types-lib/index.js [hunk 1]
```diff
@@ -100,6 +101,16 @@ export const
         return true;
       }
 
+      let customValidationResult = false;
+
+      if (validate) {
```
Comment (amourzenkov-sc) [inline src/data-types-lib/index.js]: EMPTY_FIELD_VALUE validation *must not* be skipped by custom validator

File: src/data-types-lib/index.js [hunk 2]
```diff
@@ -100,6 +101,16 @@ export const
...
+        try {
+          customValidationResult = validate(value, instance)
+        } catch (err) {
+          throw err
```
Comment (amourzenkov-sc) [inline src/data-types-lib/index.js]: According to the existing **validate()** pattern, custom validator errors should be appended to the resultant **errors** array.

File: src/data-types-lib/index.js [hunk 3]
```diff
@@ -100,6 +101,16 @@ export const
         return true;
       }
 
+      let customValidationResult = false;
+
+      if (validate) {
+        try {
+          customValidationResult = validate(value, instance)
```
Comment (amourzenkov-sc) [inline src/data-types-lib/index.js]: According to the documentation, custom **validate()** return value should be ignored.

File: src/crudeditor-lib/views/create/reducer.js [hunk 4]
```diff
@@ -244,7 +244,7 @@ export default modelDefinition => (
       }
 
       try {
-        validate(newFormValue);
+        validate(newFormValue, storeState.formInstance);
```
[the remark this answers is not in the document]
Comment (estambakio-sc) [reply src/crudeditor-lib/views/create/reducer.js]: Why?

**Comment by amourzenkov-sc** [comment]
Done by commit ec1da11

**Pull request closed without merging**